partner_ref_unique/models/res_partner.py

```diff
@@ -9,23 +9,34 @@
 class ResPartner(models.Model):
     _inherit = "res.partner"
 
-    @api.constrains("ref", "is_company", "company_id")
+    @api.constrains("ref", "is_company", "company_id", "parent_id")
     def _check_ref(self):
         for partner in self.filtered("ref"):
             # If the company is not defined in the partner, take current user company
             company = partner.company_id or self.env.company
             mode = company.partner_ref_unique
-            if mode == "all" or (mode == "companies" and partner.is_company):
-                domain = [
-                    ("id", "!=", partner.id),
-                    ("ref", "=", partner.ref),
-                ]
-                if mode == "companies":
+            if mode == "none":
+                continue
+            domain = [
+                ("id", "!=", partner.id),
+                ("ref", "=", partner.ref),
+            ]
+            if mode == "companies":
+                if partner.is_company:
                     domain.append(("is_company", "=", True))
-                other = self.search(domain)
-                # Don't raise when coming from contact merge wizard or no duplicates
-                if other and not self.env.context.get("partner_ref_unique_merging"):
-                    raise ValidationError(
-                        _("This reference is equal to partner '%s'")
-                        % other[0].display_name
-                    )
+                else:
+                    domain.append(("is_company", "=", False))
```

Comment:
Is this needed ? As it worked before and nothing has been changed for this mode.

Reply:
I need to modify the domain as I refactored the first "if" condition, the behavior for this mode didn't change, and tests stayed with the same conditions